lib/remote/actionshandler.cpp

```diff
@@ -88,7 +89,28 @@ bool ActionsHandler::HandleRequest(
 	if (params)
 		verbose = HttpUtility::GetLastParameter(params, "verbose");
 
+	std::shared_lock wgLock{*waitGroup, std::try_to_lock};
+	if (!wgLock) {
+		HttpUtility::SendJsonError(response, params, 503, "Shutting down.");
+		return true;
+	}
+
 	for (ConfigObject::Ptr obj : objs) {
+		if (!waitGroup->IsLockable()) {
+			if (wgLock) {
+				wgLock.unlock();
+			}
+			
+			Dictionary::Ptr fail = new Dictionary({
+				{ "code", 503 },
+				{ "status", "Action skipped: Shutting down."}
+			});
+
+			results.emplace_back(std::move(fail));
```

Comment:
I'd just create that dictionary in the argument of `emplace_back()`. At least I see little value in having a variable here just to move it in the next step.